lib/internal/modules/esm/assets/load.js

```diff
@@ -0,0 +1,200 @@
+const {
+  ArrayFrom,
+  ArrayPrototypeJoin,
+  ArrayPrototypePush,
+  ArrayPrototypeSlice,
+  ArrayPrototypeSplice,
+  SafeMap,
+  StringPrototypePadStart,
+  StringPrototypeSplit,
+  StringPrototypeStartsWith,
+} = primordials;
+const {
+  Buffer: {
+    concat: BufferConcat
+  }
+} = globalThis;
+const {
+  get: HTTPSGet
+} = require('https');
+const {
+  get: HTTPGet
+} = require('http');
+const {
+  promises: {
+    lookup: dnsLookup
+  }
+} = require('dns');
+/**
+ * @param {URL} parsed
+ * @returns {Promise<CacheEntry> | CacheEntry}
+ */
+function fetchWithRedirects(parsed) {
+  const existing = getCache.get(parsed.href);
+  if (existing) {
+    return existing;
+  }
+  const handler = parsed.protocol === 'http:' ? HTTPGet : HTTPSGet;
+  const result = new Promise((fulfill, reject) => {
+    const req = handler(parsed, {
+      headers: {
+        Accept: '*/*'
+      }
+    })
+    .on('error', reject)
+    .on('response', (res) => {
+      function dispose() {
+        req.destroy();
+        res.destroy();
+      }
+      if (res.statusCode >= 300 && res.statusCode <= 303) {
+        if (res.headers.location) {
+          dispose();
+          return Promise.resolve(
```

Comment:
Should use the `PromiseResolve` primordial...

Reply:
fixed